Makefile

```diff
@@ -133,38 +140,58 @@ prep:
 			mkdir -p $(INTDIR)/$(TESTS) && \
 			mkdir -p $(LIBDIR) && \
 	        mkdir -p $(BINDIR))
-	@echo directories prepared successfully.
+	-@echo directories prepared successfully.
 
 shared: $(OBJ_SHARED)
 	$(CC) -shared -o $(OUT_SHARED) $^ $(CFLAGS) $(LDFLAGS_SHARED)
-	@echo built $(OUT_SHARED) successfully.
+	-@echo built $(OUT_SHARED) successfully.
 
 static: shared
 	ar -cr $(OUT_STATIC) $(OBJ_SHARED)
-	@echo built $(OUT_STATIC) successfully.
+	-($(RANLIB) "$(OUT_STATIC)" || true) > /dev/null 2>&1
```

Comment:
The official version is "`ranlib generates an index to the contents of an archive and stores it in the archive. The index lists each symbol defined by a member of an archive that is a relocatable object.`"

On Linux and most modern systems, `ar` handles it automatically.  On some other systems, it's actually required for the static library to work (and sometimes it needs to be re-done inplace where it's installed, because the system generates some fast lookup tables for it or some other stupid reason, hence doing it at the install phase too).  Can't hurt anyway, unless you have a giant static library (like, multi-MB).

Reply:
> The official version is "`ranlib generates an index to the contents of an archive and stores it in the archive. The index lists each symbol defined by a member of an archive that is a relocatable object.`"
> 
> On Linux and most modern systems, `ar` handles it automatically. On some other systems, it's actually required for the static library to work (and sometimes it needs to be re-done inplace where it's installed, because the system generates some fast lookup tables for it or some other stupid reason, hence doing it at the install phase too). Can't hurt anyway, unless you have a giant static library (like, multi-MB).

So, on systems where `ar` writes that index, it will be a NOOP?

Reply:
Correct.